### PR TITLE
ISPN-10605 clarifying and consolidating configuration docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.jboss.org/browse/ISPN-10605

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/con_server_module_hr.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+= Hot Rod Server
+
+Use the custom Hot Rod binary protocol to access {brandname}, which allows clients to do dynamic load balancing, failover, and smart routing.
+
+//dnaro: need to productize
+** A link:http://www.infinispan.org/hotrod-clients[variety of clients] exist for this protocol.
+** If you're clients are running Java, this should be your defacto server module choice because it allows for dynamic load balancing and failover. This means that Hot Rod clients can dynamically detect changes in the topology of Hot Rod servers as long as these are clustered. When new nodes join or leave, clients update their Hot Rod server topology view. Also, when Hot Rod servers are configured with distribution, clients can detect where a particular key resides and can route requests smartly.
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/con_server_module_hr.adoc]: s/you're/your/

File: documentation/src/main/asciidoc/topics/proc_configuring_declaratively.adoc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+.Cache configuration
+Cache configuration is specific to the actual caching domain itself: it specifies eviction, locking, transaction, clustering, persistence etc.
+You can specify as many named cache configurations as you need. One of these caches can be indicated as the `default` cache,
+which is the cache returned by the `CacheManager.getCache()` API, whereas other named caches are retrieved via the `CacheManager.getCache(String name)` API.
+
+Whenever they are specified, named caches inherit settings from the default cache while additional behavior can be specified or overridden.
+{brandname} also provides a very flexible inheritance mechanism, where you can define a hierarchy of configuration templates,
+allowing multiple caches to share the same settings, or overriding specific parameters as necessary.
+
+NOTE: Embedded and Server configuration use different schemas, but we strive to maintain them as compatible as possible so that you
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/topics/proc_configuring_declaratively.adoc]: No longer true. The server configuration just uses an additional namespace for the server bits.

**Comment by tristantarrant** [comment]
And it needs rebasing

**Comment by oraNod** [comment]
> Just a couple of changes needed.

@tristantarrant should be good to go now. could you please check and merge this bad boy?

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**